### PR TITLE
docs: Update build-time dependencies for Debian/Ubuntu

### LINKED ISSUE #55: Update developer installation requirements
The [current docs mention](https://github.com/scikit-hep/fastjet/blob/f1c7241fac1d59455f8211a22557140d9534fde3/README.md#installation-for-developers) build dependencies (on Debian/Ubuntu) of

```console
apt-get install -y libmpfr-dev libboost-dev
```

but it seems from some experimenting with the `python:3.9` Docker image that the requirements on Debian are

```console
apt install -y libboost-dev gfortran swig
```

given that the following Dockerfile builds

```Dockerfile
FROM python:3.9

ENV PATH=/usr/local/venv/bin:"${PATH}"
RUN apt-get update -y && \
    apt-get install -y \
        libboost-dev \
        gfortran \
        swig && \
    apt-get -y autoclean && \
    apt-get -y autoremove && \
    rm -rf /var/lib/apt/lists/* && \
    python -m venv /usr/local/venv && \
    python -m pip --no-cache-dir install --upgrade pip setuptools wheel && \
    python -m pip --no-cache-dir install --pre fastjet && \
    python -m pip list
```
```console
docker build -f Dockerfile -t fasjet:issue-55 .
```

It should probably be pointed out that the build dependencies are for Debian/Ubuntu only.

### PR DESCRIPTION
**Pull request by matthewfeickert**
Resolves #55 

**Suggested squash and merge commit message**:
```
* Update build-time dependencies for Debian based operating systems
   - Add: gfortran swig autoconf libtool
   - Note that these are for Debian/Ubuntu
* Update build-time dependencies in CI to match those in docs
* Advocate for using `python -m pip` in all uses of `pip`
* Add CGAL to .gitignore
```

### CODE REVIEW

**Review by jpivarski — APPROVED** [review]
I almost wrote exactly the same PR; it has my approval. I'll leave it to @aryan26roy for final approval and merging.

**Comment by aryan26roy** [comment]
Everything looks good to me. Merging and closing the issue. @jpivarski  @matthewfeickert

**Review by aryan26roy — APPROVED** [review]

**Pull request merged**
